### ISSUE TITLE
Running into uncaught exceptions when trying to extract from flow

### ISSUE THREAD

**Issue by gergelyke** [atlassian/extract-react-types#41]
When trying to extract from flow, I get the following uncaught exception:

```
{ Error: (client) ../src/button/button.js (../node_modules/extract-react-types-loader!../src/button/button.js)
Module build failed (from ../node_modules/extract-react-types-loader/index.js):
TypeError: Cannot read property 'kind' of undefined
    at props.members.forEach.p ($/node_modules/extract-react-types/index.js:52:29)
    at Array.forEach (<anonymous>)
    at getPropFromObject ($/node_modules/extract-react-types/index.js:45:17)
    at getProp ($/node_modules/extract-react-types/index.js:74:12)
    at defaultProps.forEach.property ($/node_modules/extract-react-types/index.js:237:18)
    at Array.forEach (<anonymous>)
    at addDefaultProps ($/node_modules/extract-react-types/index.js:232:16)
    at convertReactComponentClass ($/node_modules/extract-react-types/index.js:263:10)
    at converters.Program ($/node_modules/extract-react-types/index.js:182:19)
    at convert ($/node_modules/extract-react-types/index.js:1433:16)
    at extractReactTypes ($/node_modules/extract-react-types/index.js:1470:10)
    at Object.extractReactTypesLoader ($/node_modules/extract-react-types-loader/index.js:65:17)
```

Repro: https://github.com/uber-web/baseui/pull/714/files#diff-6fabde4439377979ef1bfcd17147b620R43

**Comment by Noviny** [comment]
Looking into it quickly, it's struggling to resolve the defaultProps. At a guess it's the spread and the property, though it could also be a failure to grab the information from the other file.

Line 52 (or something around it) will need to be updated, but if you're looking for the fastest way forward, I'd try some different configurations for [this line](https://github.com/uber-web/baseui/blob/5b688902c33a6f30534597a1a5330bae11521235/src/button/button.js#L22) as well as just checking that removing the defaults resolves the problem you are facing.

**Comment by gergelyke** [comment]
Thanks Ben, will keep you posted!

**Comment by gergelyke** [comment]
So it seems that once I move the default props into the `button.js` file, the error goes away - any chance you can point me to where I should look for the bug in the codebase? Happy to work on a fix

**Comment by Noviny** [comment]
The error comes from line 52, which is in the [getPropFromObject](https://github.com/atlassian/extract-react-types/blob/c35845fd7e8b6e17bbc97c4198aa29b55499b01b/index.js#L34) function.

It fails because a member property doesn't have a key, and we've assumed all members in objects will have a key. I'm guessing the code at this point is a spread or an unresolved import or something?

It may be a new conditional, or an expansion of the conditional on line 46.

**Comment by Noviny** [comment]
Fixed in version 0.16.1 of extract react types and 0.5.3 of kind2string.
